vxsandbox/resources/tests/test_outbound.py

```diff
@@ -1,46 +1,341 @@
 from twisted.internet.defer import succeed, inlineCallbacks
 
+from vumi.errors import InvalidEndpoint
+
 from vxsandbox.resources.outbound import OutboundResource
 from vxsandbox.resources.tests.utils import ResourceTestCaseBase
+from vxsandbox.tests.utils import DummyAppWorker
+
+
+class StubbedAppWorker(DummyAppWorker):
+
+    class DummyApi(DummyAppWorker.DummyApi):
+        def __init__(self, inbound_messages):
+            self._inbound_messages = inbound_messages
+
+        def get_inbound_message(self, msg_id):
+            return self._inbound_messages.get(msg_id)
+
+    sandbox_api_cls = DummyApi
+
+    def __init__(self):
+        super(StubbedAppWorker, self).__init__()
+        self._inbound_messages = {}
+        self._valid_endpoints = set(['default'])
+
+    def check_endpoint(self, endpoint):
+        if endpoint not in self._valid_endpoints:
+            raise InvalidEndpoint(
+                "Endpoint %r is not configured" % (endpoint,))
+
+    def create_sandbox_api(self):
+        return self.sandbox_api_cls(self._inbound_messages)
+
+    def add_inbound_message(self, msg):
+        self._inbound_messages[msg['message_id']] = msg
+
+    def add_endpoint(self, endpoint):
+        self._valid_endpoints.add(endpoint)
 
 
 class TestOutboundResource(ResourceTestCaseBase):
 
+    app_worker_cls = StubbedAppWorker
     resource_cls = OutboundResource
 
     @inlineCallbacks
     def setUp(self):
         super(TestOutboundResource, self).setUp()
         yield self.create_resource({})
+        self.app_worker.add_inbound_message({u'message_id': u'known-id'})
+
+    @inlineCallbacks
+    def assert_cmd_fails(self, reason, cmd, resource_config=None, **cmd_args):
+        yield self.create_resource(resource_config or {})
+        reply = yield self.dispatch_command(cmd, **cmd_args)
+        self.check_reply(reply, success=False, reason=reason)
+        self.assertFalse(self.app_worker.mock_calls['send_to'], [])
+        self.assertFalse(self.app_worker.mock_calls['reply_to'], [])
+        self.assertFalse(self.app_worker.mock_calls['reply_to_group'], [])
```

Comment:
Should this not be `assertEqual`? I'm also not sure what these 3 lines are testing.